aten/src/ATen/native/quantized/cpu/qmul.cpp

```diff
@@ -50,22 +51,58 @@ Tensor _mul_out(Tensor& out, const Tensor& self, const Tensor& other) {
 
 template <bool ReLUFused = false>
 Tensor _mul_scalar_out(Tensor& out, const Tensor& self, Scalar other) {
-  int64_t zero_point = out.q_zero_point();
-  double scale = out.q_scale();
   int64_t self_zero_point = self.q_zero_point();
   double self_scale = self.q_scale();
+  double other_val = other.toDouble();
+
+  double scale_prime;
+  int64_t zero_point_prime;
+
+  AT_DISPATCH_QINT_TYPES(out.scalar_type(), "qmul_scalar", [&]() {
+    int64_t q_min = std::numeric_limits<underlying_t>::min();
+    int64_t q_max = std::numeric_limits<underlying_t>::max();
+
+    if (other_val > 0.0) {
+      scale_prime = other_val * self_scale;
+      zero_point_prime = self_zero_point;
 
-  auto iter = TensorIterator::unary_op(out, self);
-  AT_DISPATCH_QINT_TYPES(out.scalar_type(), "qmul", [&]() {
-    cpu_kernel(iter, [&](scalar_t a) -> scalar_t {
-      const auto da = at::dequantize_val(self_scale, self_zero_point, a);
-      float c = da * other.toFloat();
       if (ReLUFused) {
-        c = std::max<float>(c, 0.0);
+        qrelu_stub(self.device().type(), self, out);
+      } else {
+        out.copy_(self);
       }
-      return at::quantize_val<scalar_t>(scale, zero_point, c);
-    });
+      out.set_quantizer_(make_per_tensor_affine_quantizer(
+          scale_prime, zero_point_prime, self.scalar_type()));
+    } else if (other_val == 0.0) {
+      scale_prime = 1.0;
+      zero_point_prime = 0;
+
+      // Strided "memset"
+      // Set all values to 0
+      auto iter = TensorIterator::unary_op(out, self);
+      cpu_kernel_vec(
+          iter,
+          [&](scalar_t a) -> scalar_t { return scalar_t(0); },
+          [&](Vec256<scalar_t> vec) -> Vec256<scalar_t> {
+            return Vec256<scalar_t>(scalar_t(0));
+          });
+      out.set_quantizer_(make_per_tensor_affine_quantizer(
+          scale_prime, zero_point_prime, self.scalar_type()));
+    } else /* other_val < 0.0 */ {
+      scale_prime = other_val * self_scale;
+      zero_point_prime = q_max - (self_zero_point - q_min) - 1;
```

Comment:
zero_point_prime should be = q_max -(self_zero_point - q_min). There should be no -1.